src/log4stash/ElasticSearchAppender.cs

```diff
@@ -176,24 +178,40 @@ protected override void Append(LoggingEvent loggingEvent)
                 return;
             }
 
-            if (DropEventsOverBulkLimit && _bulk.Count >= BulkSize)
+            if (HasOverflowState())
             {
-                _tolerateCalls.Call(() =>
-                    _eventWriter.Warn(GetType(),
-                        "Message lost due to bulk overflow! Set DropEventsOverBulkLimit to false in order to prevent that."),
-                    GetType(), 0);
+                ReportOverflow();
                 return;
             }
 
             var logEvent = _logEventConverter.ConvertLogEventToDictionary(loggingEvent);
             PrepareAndAddToBulk(logEvent);
 
-            if (!DropEventsOverBulkLimit && _bulk.Count >= BulkSize && BulkSize > 0)
+            if (HasOverflowState())

```

Comment:
This check is redundant since you have it earlier.

I suppose you meant to check if `DropEventsOverBulkLimit && !TotalDropEventLimit.HasValue`
Although I'm not sure